src/Startup.cs

```diff
@@ -37,22 +37,10 @@ public void ConfigureServices(IServiceCollection services)
 
             services.AddDbContext<PiggyDbContext>(options =>
                 options.UseSqlServer(Configuration.GetConnectionString("DefaultConnection")));
-            services.AddScoped(provider =>
-            {
-                var db = provider.GetService<PiggyDbContext>();
-                return new UserDatabase(db);
-            });
-            services.AddScoped(provider =>
-            {
-                var db = provider.GetService<PiggyDbContext>();
-                return new EntryDatabase(db);
-            });
-            services.AddScoped(provider =>
-            {
-                var db = provider.GetService<PiggyDbContext>();
-                return new GoalDatabase(db);
-            });
-            services.AddScoped<EmailSender>();
+            services.AddScoped<UserDatabase>();
+            services.AddScoped<EntryDatabase>();
+            services.AddScoped<GoalDatabase>();
+            services.Configure<EmailSender>(options => Configuration.GetSection("Email").Bind(options));
```

Comment:
So he know which data to read as options by section? And then assigns it to same name properties?

Reply:
Basically that's how it works, you can read more here: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-5.0